lib/fuse_loop_mt.c

```diff
@@ -155,7 +155,7 @@ static void *fuse_do_work(void *data)
 
 		if (!isforget)
 			mt->numavail--;
-		if (mt->numavail == 0)
+		if (mt->numavail == 0 && mt->numworker < mt->max_idle)
```

Comment:
This seems to contradict the description of the option - max *idle* threads. We are not creating an idle thread here, but a busy one.

Reply:
I do not think it's a good idea to change the meaning of max_idle (or to drop it) for backwards compatibility reason.

Introducing a new `max_threads` variable seems fine to me. I do not think that introducing a default limit would cause issues with backwards compatibility. I'm less sure about what's a reasonable value though - any suggestions?

(At the same time, I think we should default `max_idle` to max_threads to prevent the expensive destruction/creation cycle by default).

Reply:
Ok, fine with max_threads. I would set it by default to 10 - same default as 

. But now we cannot simply add another variable to struct fuse_loop_config(), because it is exposed to everything and not versioned - making it larger would cause memory access beyond struct for existing applications not recompiled against the new libfuse. I.e, a new fuse_session_loop_mt_33 is needed. At least from my point of view that also allows to deprecate the max_idle_threads variable.

Reply:
Hmm, so this is going to be rather big. I stop for now and will finish it in the morning. Posting an unfinished patch here so that you can get an idea (logic part completed, but not even compilation tested...).

Reply:
Please let me know what you think about this. I increased the version as this introduces a new API.